components/RoomList/index.js

```diff
@@ -1,37 +1,44 @@
 import { useState, useEffect } from "react";
 import RoomButton from "../RoomButton";
-import { RoomListContainer, RoomListItem, BasicListItem } from "./styles";
+import {
+  RoomListContainer,
+  RoomListItem,
+  BasicListItem,
+  ButtonContainer,
+} from "./styles";
 import Divider from "../Divider";
 import RoomIcon from "../RoomIcon";
+import Link from "next/link";
+import EditButton from "../EditButton";
 
 export default function RoomList({ userType }) {
   const [rooms, setRooms] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
-  useEffect(() => {
-    async function fetchRoomData() {
-      setIsLoading(true);
-      try {
-        const response = await fetch("/api/rooms");
-        if (!response.ok) {
-          throw new Error("Failed to fetch room data");
-        }
-        const data = await response.json();
-        setRooms(data);
-        setIsLoading(false);
-      } catch (error) {
-        console.error("Error fetching room data:", error);
-        setIsLoading(false);
+  async function fetchRoomData() {
+    setIsLoading(true);
+    try {
+      const response = await fetch("/api/rooms");
+      if (!response.ok) {
+        throw new Error("Failed to fetch room data");
       }
+      const data = await response.json();
+      setRooms(data);
+      setIsLoading(false);
+    } catch (error) {
+      console.error("Error fetching room data:", error);
+      setIsLoading(false);
     }
+  }
 
+  useEffect(() => {
     fetchRoomData();
   }, []);
```

Comment:
Why did you move the function outside the useEffect callback? Technically, when the function is _outside_, you need to pass it as a dependency into the dependency array. When it's inside, you don't. Better move it back into the callback.